platform/src/main/java/org/hillview/SamplingBenchmark.java

```diff
@@ -0,0 +1,104 @@
+package org.hillview;
+
+import org.hillview.dataset.LocalDataSet;
+import org.hillview.dataset.api.IDataSet;
+import org.hillview.dataset.api.ISketch;
+import org.hillview.sketches.BucketsDescriptionEqSize;
+import org.hillview.sketches.Histogram;
+import org.hillview.sketches.HistogramSketch;
+import org.hillview.table.Table;
+import org.hillview.table.api.ColumnAndConverterDescription;
+import org.hillview.table.api.IColumn;
+import org.hillview.table.api.ITable;
+import org.hillview.table.columns.DoubleArrayColumn;
+import org.hillview.table.membership.DenseMembershipSet;
+import org.hillview.table.membership.FullMembershipSet;
+import org.hillview.table.membership.SparseMembershipSet;
+import org.hillview.utils.HillviewLogger;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Level;
+
+public class SamplingBenchmark {
+    public static void main(String[] args) throws IOException, InterruptedException {
+        // Testing the performance of histogram computations
+        final int bucketNum = 40;
+        final int mega = 1024 * 1024;
+        final int runCount = 10;
+        HillviewLogger.instance.setLogLevel(Level.OFF);
+        BucketsDescriptionEqSize buckDes = new BucketsDescriptionEqSize(0, 100, bucketNum);
+        PrintWriter writer = new PrintWriter("SamplingBenchmarkFull", "UTF-8");
+
+        for (int elements = 10 * mega; elements < 101 * mega; elements += (10 * mega)) {
+            final DoubleArrayColumn col = HistogramBenchmark.generateDoubleArray(elements, 100);
+            ITable table = createTable(elements, col);
+            writer.println("averaging " + runCount + " runs, with " + elements + " elements");
+            writer.println("rate,  elements, min, max, mean (ms)");
+            final IDataSet<ITable> ds = new LocalDataSet<ITable>(table, false);
+            for (double rate = 0.01; rate < 0.2; rate += 0.01) {
+                ISketch<ITable, Histogram> sk = new HistogramSketch(
+                        buckDes, new ColumnAndConverterDescription(col.getName()), rate, 0);
+                Runnable r = () -> ds.blockingSketch(sk);
+                writer.print(rate + " , ");
+                runNTimes(r, runCount, "Dataset histogram wo sampling", elements, writer);
+            }
+            ISketch<ITable, Histogram> sk = new HistogramSketch(
+                    buckDes, new ColumnAndConverterDescription(col.getName()), 1, 0);
+            Runnable r = () -> ds.blockingSketch(sk);
+            writer.print("1 , ");
+            runNTimes(r, runCount, "Dataset histogram wo sampling", elements, writer);
+            System.out.println(elements);
+        }
+        writer.close();
+    }
+
+    private static ITable createTable(final int colSize, final IColumn col) {
```

Comment:
Can you share these methods with the previous class?
If they are tests it may be even easier.

Reply:
No, there are slight differences in the implementaiton